### PR TITLE
Add auto build conf

### PR DESCRIPTION
**Pull request by perlitz**
The pull request adds a method that builds a configuration given the evaluation budget

### CODE REVIEW

File: build_run_specs_full.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+if __name__ == "__main__":
+
+    import pandas as pd
+    df = pd.DataFrame(entries)
+    scenarios = df.value_counts('scenario').to_dict()
+
+    total_n_examples = 500
+    max_eval_instances_per_scenario = total_n_examples//len(df.scenario.unique())
+
+    for scenario, n_sucscenarios in scenarios.items():
```
Comment (artidoro) [inline build_run_specs_full.py]: Just a typo here.

File: build_run_specs_full.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+
+if __name__ == "__main__":
+
+    import pandas as pd
+    df = pd.DataFrame(entries)
+    scenarios = df.value_counts('scenario').to_dict()
+
+    total_n_examples = 500
+    max_eval_instances_per_scenario = total_n_examples//len(df.scenario.unique())
```
Comment (artidoro) [inline build_run_specs_full.py]: You will have fewer than `total_n_examples` if the division has a remainder, which is the case for both 500 and 600. 
For example, when running with `total_n_examples` of 600, you will end up with 570 total examples. Maybe it's worth distributing remainder somehow like you were doing earlier, or change the name of the file to the actual number of examples that will be run.

**Review by artidoro — APPROVED** [review]
Some small comments but overall I think this will be useful to speed up eval time.

File: build_run_specs_full.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    {'scenario':'misconceptions_russian','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=misconceptions_russian,subtask=", 'priority': 1},
+
+    # 15. novel_concepts: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/novel_concepts
+    {'scenario':'novel_concepts','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=novel_concepts,subtask=", 'priority': 1},
+
+    # 16. operators: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/operators
+    {'scenario':'operator','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=operators,subtask=", 'priority': 1},
+
+    # 17. parsinlu_reading_comprehension: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/parsinlu_reading_comprehension
+    {'scenario':'parsinlu_reading_comprehension','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=parsinlu_reading_comprehension,subtask=", 'priority': 1},
```
Comment (weiweiy) [inline build_run_specs_full.py]: I propose remove this Persian language task

File: build_run_specs_full.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    {'scenario':'logic_grid_puzzle','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=logic_grid_puzzle,subtask=", 'priority': 1},
+
+    # 13. logical_deduction: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/logical_deduction
+    {'scenario':'logical_deduction','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=logical_deduction,subtask=three_objects", 'priority': 1},
+    {'scenario':'logical_deduction','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=logical_deduction,subtask=five_objects", 'priority': 1},
+    {'scenario':'logical_deduction','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=logical_deduction,subtask=seven_objects", 'priority': 1},
+
+    # 14. misconceptions_russian: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/misconceptions_russian
+    {'scenario':'misconceptions_russian','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=misconceptions_russian,subtask=", 'priority': 1},
```
Comment (weiweiy) [inline build_run_specs_full.py]: Remove Russian Specific task

File: build_run_specs_full.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    {'scenario':'conlang_translation','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conlang_translation,subtask=vaomi_to", 'priority': 1},
+
+    # 6. emoji_movie: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/emoji_movie
+    {'scenario':'emoji_movie','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=emoji_movie,subtask=", 'priority': 1},
+
+    # 7. formal_fallacies_syllogisms_negation: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/formal_fallacies_syllogisms_negation
+    {'scenario':'formal_fallacies_syllogisms_negation','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=formal_fallacies_syllogisms_negation,subtask=", 'priority': 1},
+
+    # 8. hindu_knowledge: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/hindu_knowledge
+    {'scenario':'hindu_knowledge','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=hindu_knowledge,subtask=", 'priority': 1},
```
Comment (weiweiy) [inline build_run_specs_full.py]: Remove Hindu specific task

File: build_run_specs_full.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=contradictions", 'priority': 1},
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=emergent_properties", 'priority': 1},
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=fanciful_fictional_combinations", 'priority': 1},
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=homonyms", 'priority': 1},
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=invented_words", 'priority': 1},
+
+    # 5. conlang_translation: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/conlang_translation
```
Comment (weiweiy) [inline build_run_specs_full.py:16]: @artidoro what your feelings on these Conlang tasks?

File: build_run_specs_full.py [hunk 7]
```diff
@@ -0,0 +1,237 @@
+entries = [
+    #bigbench
+    # 1. auto_debugging: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/auto_debugging
```
Comment (weiweiy) [inline build_run_specs_full.py:3]: @msaroufim  @artidoro, how you feel about code understanding type of tasks for auto_debugging and code description task?

**Comment by weiweiy** [comment]
@perlitz , Thank you for creating this PR!
In general, I propose to remove Hindu, Russian and Persian specific tasks... however I am somewhat tore when it comes to code understanding (auto_debugging / code_line_description) and Colang tasks.  
@artidoro @jisaacso @msaroufim  thoughts?

**Comment by msaroufim** [comment]
personally i think the coding and conlang tasks are interesting

File: build_run_specs_full.py (hunk 1 above)
Comment (perlitz) [reply build_run_specs_full.py]: will fix

File: build_run_specs_full.py (hunk 2 above)
Comment (perlitz) [reply build_run_specs_full.py]: good idea, I will distribute these

File: build_run_specs_full.py (hunk 3 above)
Comment (perlitz) [reply build_run_specs_full.py]: will do

File: build_run_specs_full.py (hunk 4 above)
Comment (perlitz) [reply build_run_specs_full.py]: will do

File: build_run_specs_full.py (hunk 5 above)
Comment (perlitz) [reply build_run_specs_full.py]: will do

**Review by weiweiy — APPROVED** [review]
@perlitz  Thanks again for making this PR and addressing our issues.

**Pull request merged**